topics/releases.md

```diff
@@ -1,81 +1,103 @@
 
-At some point, when a given Redis release is in the Release Candidate state
-for enough time, we observe that the frequency at which critical bugs are
-signaled starts to decrease, to the point that for a few weeks we don't have
-any serious bugs reported.
+Patches primarily consist of bug fixes and never introduce any compatibility issues.
 
-When this happens, the release is marked as stable.
+Upgrading from a previous patch-level version is always safe and seamless.
 
-Version numbers
+New features and configuration directives may be added, or default values changed, as long as these don’t carry significant impacts or introduce operations-related issues.
+
+### Minor Versions
+
+Minor versions usually deliver maturity and extended functionality.
+
+Upgrading between minor versions does not introduce any application-level compatibility issues.
+
+Minor releases may include new commands and data types that introduce operations-related incompatibilities, including changes in data persistence format and replication protocol.
+
+### Major Versions
+
+Major versions introduce new capabilities and significant changes.
+
+Ideally, these don't introduce application-level compatibility issues.
+
+Release Schedule
 ---
 
-Stable releases follow the usual `major.minor.patch` versioning schema, with the following special rules:
+A new major version is planned for release at the beginning of every year.
+
+Generally, every new major release is followed by a minor version after six months.
+
+Patches are released to fix high-urgency issues, or once a stable version accumulates enough fixes to justify it.
 
-* The minor is even in stable versions of Redis.
-* The minor is odd in unstable, development, frozen, release candidates. For instance the unstable version of 2.8.x will have a version number in the form 2.7.x. In general the unstable version of x.y.z will have a version x.(y-1).z.
-* As an unstable version of Redis progresses, the patch level is incremented from time to time, so at a given time you may have 2.7.2, and later 2.7.3 and so forth. However when the release candidate state is reached, the patch level starts from 101. So for instance 2.7.101 is the first release candidate for 2.8, 2.7.105 is Release Candidate 5, and so forth.
+For contacting the core team on sensitive matters and security issues, please email [redis@redis.io](mailto:redis@redis.io).
 
 Support
 ---
 
-Older versions are not supported as we try very hard to make the
-Redis API mostly backward compatible. Upgrading to newer versions
-is usually trivial.
+As a rule, older versions are not supported as we try very hard to make the Redis API mostly backward compatible.
+
+Upgrading to newer versions is the recommended approach and is usually trivial.
+
+The last minor version of the latest major version is fully supported.
+
+The previous minor version of the current major version and the latest minor version of the previous major version are partially supported.
```

Comment:
maybe this should come with some example to make it easier to comprehend

Reply:
Tried adding an example and making it slightly clearer :)